src/mcts/search.h

```diff
@@ -57,7 +57,9 @@ class Search {
          BestMoveInfo::Callback best_move_callback,
          ThinkingInfo::Callback info_callback, const SearchLimits& limits,
          const OptionsDict& options, NNCache* cache,
-         SyzygyTablebase* syzygy_tb);
+         SyzygyTablebase* syzygy_tb,
+         std::chrono::steady_clock::time_point start_time =
```

Comment:
There is a zoo of various start times now...
What do you think about doing this instead:
1. Replace `SearchLimits::time_ms` with `SearchLimits::search_deadline`, make it `std::optional<chrono::steady_clock::time_point>`, and populate in `EngineController::PopulateSearchLimits` (and pass start_time to that function)
2. Make `nps_start_time_` also std::optional and set it after the first iteration which had real NN eval.

Reply:
`SearchLimits::time_ms` is also used for selfplay with `--movetime=x` so it isn't easy to remove completely.

Reply:
There we can also compute end time in game.cc and pass it.